messenger.go

```diff
@@ -80,10 +87,18 @@ func WithChats(
 	}
 }
 
+func WithGenericDiscoveryTopicSupport() func(c *config) error {
+	return func(c *config) error {
+		c.featureFlags.genericDiscoveryTopicEnabled = true
+		return nil
+	}
+}
+
 func NewMessenger(
 	identity *ecdsa.PrivateKey,
 	server transport.Server,
 	shh *whisper.Whisper,
+	logger *zap.Logger,
```

Comment:
Just noticed your comment :) Maybe we can put this into `opts ...Option`? As I mentioned, it's not crucial business logic. Also if not provided, we should default to some standard one. I guess I can pass `nil` now and it will panic.

Reply:
We could provide a default one, but AFAICT there's no method to shutdown the messanger, where we'd be able to call `logger.Sync()`. Is that something we want to add?

Reply:
OK, I've added a `Shutdown` method to Messenger with a list of `shutdownTasks` that we register to be run in `Shutdown`. That avoids keeping a pointer to `Logger` in `Messenger`.